packages/kas/src/nodes.ts

```diff
@@ -1203,29 +1227,30 @@ export class Mul extends Seq {
             // TODO(alex): combine if all inverses are the same e.g. ln(y)*ln(z)/ln(x)/ln(x)
         }
 
-        pairs = trigs.concat(logs).concat(exprs);
-
-        var collected = _.map(pairs, function (pair) {
+        var collected = _.map([...trigs, ...logs, ...exprs], (pair) => {
             return new Pow(pair[0], pair[1]).collect(options);
         });
 
         return new Mul([number].concat(collected)).flatten();
     }
 
     isSubtract() {
-        return _.any(this.terms, function (term) {
-            return term instanceof Num && term.hints.subtract;
+        return _.any(this.terms, (term) => {
+            return term instanceof Num && Boolean(term.hints.subtract);
         });
     }
 
     // factor a single -1 in to the Mul
     // combine with a Num if all Nums are positive, else add as a term
-    factorIn(hint) {
+    factorIn(hint: keyof Hints): Expr {
         var partitioned = this.partition();
-        var numbers = partitioned[0].terms;
+        // `partition` splits the terms into two Seqs - one containing
+        // only Nums and the all non-Num nodes.
+        var numbers = partitioned[0].terms as Num[];
         var fold =
             numbers.length &&
-            _.all(numbers, function (num) {
+            _.all(numbers, (num) => {
+                // TODO(kevinb): this is unsafe
```

Comment:
I think this comment could use more context. Is there something we need to change? Something we need to look out for?

Reply:
I think this comment is stale.  I'll remove it.